src/middlewared/middlewared/plugins/pool_/replace_disk.py

```diff
@@ -45,78 +44,57 @@ async def replace(self, job, oid, options):
-
-        create_swap = found[0] in ('data', 'spare')
+        verrors.check()
 
         swap_disks = [disk['devname']]
-        # If the disk we are replacing is still available, remove it from swap as well
-        if found[1] and os.path.exists(found[1]['path']):
-            from_disk = await self.middleware.call(
-                'disk.label_to_disk', found[1]['path'].replace('/dev/', '')
-            )
-            if from_disk:
+        if found[1] and await self.middleware.run_in_thread(os.path.exists, found[1]['path']):
+            if from_disk := await self.middleware.call('disk.label_to_disk', found[1]['path'].replace('/dev/', '')):
+                # If the disk we are replacing is still available, remove it from swap as well
                 swap_disks.append(from_disk)
 
         await self.middleware.call('disk.swaps_remove_disks', swap_disks)
 
         vdev = []
-        await self.middleware.call(
-            'pool.format_disks', job, {disk['devname']: {'vdev': vdev, 'create_swap': create_swap}},
-        )
+        format_opts = {disk['devname']: {'vdev': vdev, 'create_swap': found[0] in ('data', 'spare')}}
+        await self.middleware.call('pool.format_disks', job, format_opts)
 
-        new_devname = vdev[0].replace('/dev/', '')
-
-        job.set_progress(30, 'Replacing disk')
         try:
-            await self.middleware.call(
-                'zfs.pool.replace', pool['name'], options['label'], new_devname
-            )
-            # If we are replacing a faulted disk, kick it right after replace
-            # is initiated.
+            job.set_progress(30, 'Replacing disk')
+            new_devname = vdev[0].replace('/dev/', '')
+            await self.middleware.call('zfs.pool.replace', pool['name'], options['label'], new_devname)
             try:
-                vdev = await self.middleware.call(
-                    'zfs.pool.get_vdev', pool['name'], options['label'],
-                )
+                vdev = await self.middleware.call('zfs.pool.get_vdev', pool['name'], options['label'])
                 if vdev['status'] not in ('ONLINE', 'DEGRADED'):
+                    job.set_progress(80, 'Detaching old disk')
+                    # If we are replacing a faulted disk, kick it right after replace is initiated.
                     await self.middleware.call('zfs.pool.detach', pool['name'], options['label'])
             except Exception:
-                self.logger.warn('Failed to detach device', exc_info=True)
+                self.logger.warning('Failed to detach device with label %r', options['label'], exc_info=True)
         finally:
             # Needs to happen even if replace failed to put back disk that had been
             # removed from swap prior to replacement
             asyncio.ensure_future(self.middleware.call('disk.swaps_configure'))
 
-        if old_disk:
-            await self.middleware.call('disk.copy_settings', old_disk, disk)
+        if options['preserve_settings']:
```

Comment:
As we are changing logic here, should we add a test in order to ensure this is still working as expected?